src/common/snippets/src/utils/tokenization_utils.cpp

```diff
@@ -0,0 +1,434 @@
+                        if (auto to_replace_with = ov::as_type_ptr<op::Subgraph>(subgraph->get_input_node_shared_ptr(i))) {
+                            // todo: In principle, we can still attach the node to the subgraph if cyclic dependency is introduced during ternary merge.
+                            //  Need to support.
+                            if (cyclicDependencyIsIntoduced(to_replace_with, currentTopoBounds))
+                                return abort_with_strategy("Attempt to perform recurrent merge for cyclic-dependent subgraphs. Aborting.");
+                            for (const auto& output : internal_consumers) {
+                                    for (auto consumer : output.get_target_inputs()) {
+                                        auto other_body = clones[subgraph->get_input_node_shared_ptr(i)];
+                                        auto other_body_result = other_body->get_results()[consumer.get_source_output().get_index()];
+                                        auto result_producer = other_body_result->input(0).get_source_output();
+
+                                        consumer.replace_source_output(result_producer.get_node_shared_ptr());
+                                    }
+                                }
+                        } else {
+                            external_inputs.push_back(subgraph->input_value(i));
+                            body_parameters.push_back(input_body_parameters[i]);
+                        }
+                    } else {
+                        external_inputs.push_back(subgraph->input_value(i));
+                        body_parameters.push_back(input_body_parameters[i]);
+                    }
+                }
+            }
+
+            // this is there stitching happens, get result of a copy of a body of currently processed input and put it to the new inputs
+            // internal output index == external output index
+            auto& input_body = clones[input_node];
+            size_t source_output_index = input_value.get_index();
+            auto source_result = input_body->get_results()[source_output_index];
+
+            // We cannot add new node, that is not Convert, after Convert (that is start node) to avoid arithmetic problems with conversion
+            // We can add any new node in Subgraph after Convert (bacause after Input)
+            //              Parameter
+            //                  |
+            //               Convert
+            //
+            // We cannot add new node, that isn't Convert, in Subgraph after existing Convert
+            //              Parameter
+            //                Relu
+            //               Convert
+            //
+            // But we can add new Convert in Subgraph after existing Convert
+            //              Parameter
+            //                Relu
+            //               Convert
+            //               Convert
+            //
```

Comment:
It looks like this logic is obsolete, because we should be able to support Converts everywhere. Do you think we can remove it?
CC @a-sidorova, we discussed it offline

Reply:
I think that we can indeed remove this logic. However, since this code is sensitive and affect tokenization of all models, I especially tried to avoid any changes in this helper.
I suggest merging this PR without the proposed changes, and do it in a separate PR which will contain only this change: this would save our time on root-cause investigation in case of unexpected degradations

Reply:
Ok, fair enough

Reply:
The PR is created: #26843